modules/Header/ui/CategoriesNav/CategoriesNavDesktop.tsx

```diff
@@ -34,6 +34,19 @@ export function CategoriesNavDesktop({
         (translatedCategory) => !getCategory(translatedCategory)?.is_featured,
     );
 
+    const [currentWidth, setCurrentWidth] = useState(0);
+    useEffect(() => {
+        setCurrentWidth(document.body.offsetWidth);
+        const handleResize = () => {
+            setCurrentWidth(document.body.offsetWidth);
+        };
+        window.addEventListener('resize', handleResize);
+
+        return () => {
+            window.removeEventListener('resize', handleResize);
+        };
+    }, []);
```

Comment:
This whole logic is related to `BlockPageScroll`, so let's move it there, that would also eliminate the need for passing `currentWidth` to `BlockPageScroll`.

Reply:
I have refactored the solution so it does not need the state anymore.  I have also eliminated the event listener for the resize event. 
I stored the value of scrollbar width inside a ref as it is only required to be computed once. This way I was able to prevent any unneccessary rerenders. 

One thing to note here is , the `scrollbarWidth ` has to be passed as a prop to `BlockPageScroll` as the `scrollbar` is not present inside the `BlockPageScroll` component and the value of its width cannot be computed from within.